File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <math.h>
using namespace std;

int solution(int n) {
    int answer = 0;
    	vector<int> num = { 2 };

	for (int i = 3; i <= n; i++)
	{
		int cnt = 0;
		bool chk = false;
		for (auto r : num)
		{
			if ((int)pow(r,2) > n) break;

			if (i % r == 0)
			{
				chk = false;
				break;
			}
			else
			{
				chk = true;
			}
		}
		if (chk) num.emplace_back(i);
	}
    return num.size();
}

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.cs
using System;
using System.Linq;
using System.Collections.Generic;
public class Solution {
    public int solution(int n) {
        int answer = 0;
        if(n == 2) return 1;
        else if(n == 3) return 2;
        List<int> all = new List<int>();
        all.Add(2);
        all.Add(3);
        for(int i = 4; i<=n;i++)
        {
            bool chk = true;
            for(int j = 0; j<all.Count;j++)
            {
                if(all[j] > Math.Sqrt(i))
                {
                    chk = true;
                    break;
                }
                if(i%all[j] == 0)
                {
                    chk = false;
                    break;
                }
            }
            if(chk) all.Add(i);
        }
        answer = all.Count;
        return answer;
    }
}